### PR TITLE
test: added tests mentioned in issue #305

### LINKED ISSUE #305: Write 3 tests for Create a Task API
## Description
As a developer for backend api,
I need cover most part of the logic of backend,
so that I can be confident that the app works as it is supposed to.

Write 3 tests that cover the following test cases for the POST /mentorship_relation/{request_id}/task API, within the tests/tasks/test_api_create_task.py file:
- User not involved in mentorship relation tries to create a task (FAIL)
- Valid user tries to create task without description in the body (FAIL)
- Valid user tries to create a task with authentication token expired (FAIL)

Make sure to add the test in the appropriate [tests/tasks/test_api_create_task.py](https://github.com/systers/mentorship-backend/blob/develop/tests/tasks/test_api_create_task.py) file.

_this issue is intended for the Google Code-in participants, it can be released after the program ends on January 23rd_

### Update [Required]
- [ ] Write 3 tests that cover 3 mentioned test cases for the Create task API

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by Goutam192002**
### Description
Added tests mentioned in issue #305 

Fixes #305 

### Type of Change:

- Code
- Quality Assurance

### How Has This Been Tested?
To check if the user is trying to create a task on a unaccepted state relation. I choose a task which is in unaccepted state and try to create a task. It returns a 400.
To check if the user is trying to create a task with empty description. I try to create a task with no description which again returns a 400.
To check if the user is trying to create a task with an expired auth token. I create a auth header in which the token has already expired. In this case the api returns a 401.

### Checklist:
**Delete irrelevant options.**

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**
- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: tests/tasks/tasks_base_setup.py [hunk 1]
```diff
@@ -31,15 +31,22 @@ def setUp(self):
             password=user2['password'],
             terms_and_conditions_checked=user2['terms_and_conditions_checked']
         )
-
+        self.third_user = UserModel(
```
Comment (Tlazypanda) [inline tests/tasks/tasks_base_setup.py]: Changes are not required in this file.

File: tests/tasks/test_api_create_task.py [hunk 2]
```diff
@@ -16,6 +17,39 @@ def test_create_task_api_resource_non_auth(self):
         self.assertEqual(401, actual_response.status_code)
         self.assertDictEqual(expected_response, json.loads(actual_response.data))
 
+    def test_user_involved_in_mentorship_relation_creates_task(self):
+        auth_header = get_test_request_header(self.second_user.id)
+
+        actual_response = self.client.post('/mentorship_relation/3/task',
+                                           follow_redirects=True, headers=auth_header, content_type='application/json',
+                                           data=json.dumps(dict(description=self.test_description)))
+        expected_response = messages.UNACCEPTED_STATE_RELATION
```
Comment (Tlazypanda) [inline tests/tasks/test_api_create_task.py]: message returned should be `MENTORSHIP_RELATION_DOES_NOT_EXIST`

File: tests/tasks/test_api_create_task.py [hunk 3]
```diff
@@ -16,6 +17,39 @@ def test_create_task_api_resource_non_auth(self):
         self.assertEqual(401, actual_response.status_code)
         self.assertDictEqual(expected_response, json.loads(actual_response.data))
 
+    def test_user_involved_in_mentorship_relation_creates_task(self):
```
Comment (Tlazypanda) [inline tests/tasks/test_api_create_task.py]: rename method to `test_user_not_involved_in_mentorship_relation_creates_task`

File: tests/tasks/test_api_create_task.py [hunk 4]
```diff
@@ -16,6 +17,39 @@ def test_create_task_api_resource_non_auth(self):
...
+        auth_header = get_test_request_header(self.second_user.id)
+
+        actual_response = self.client.post('/mentorship_relation/3/task',
+                                           follow_redirects=True, headers=auth_header, content_type='application/json',
+                                           data=json.dumps(dict(description=self.test_description)))
+        expected_response = messages.UNACCEPTED_STATE_RELATION
+
+        self.assertEqual(400, actual_response.status_code)
```
Comment (Tlazypanda) [inline tests/tasks/test_api_create_task.py]: Status code returned should be 404

File: tests/test_data.py [hunk 5]
```diff
@@ -23,3 +23,11 @@
     "password": "user2_pwd",
     "terms_and_conditions_checked": True
 }
+
+user3 = {
```
Comment (Tlazypanda) [inline tests/test_data.py]: Changes not required here

**Review by Tlazypanda — CHANGES_REQUESTED** [review]
@Goutam192002 Some changes required on the pr :)

**Comment by Goutam192002** [comment]
Done 👍

**Review by Tlazypanda — APPROVED** [review]
Approved :)

**Comment by isabelcosta** [comment]
Thank you so much for your contribution @Goutam192002  🎉 
I'm closing this PR because we decided to merge #310

**Pull request closed without merging**